### PR TITLE
Use git branch and revision in version string

### LINKED ISSUE #3089: Dev-builds better be more verbose about their version than plain "2.1.2"
I was asked to try a dev-build rather than release, and here is what i see in the Above window:

2023-02-02 03_22_54-About Cutter

"Version 2.1.2" is good for release builds - it's simplicity is a merit there.

But it is obfuscating things when it is shows so in non-stable builds!!!

This specific ZIP i got from https://ci.appveyor.com/project/rizinorg/cutter/branch/dev/artifacts whnere it is advertized as `2.1.2-git-1211` not plain `2.1.2`
It also has a commit hash, namely `baeffba7d1b2f783c06cc3af51ba7533b3f6df15`

I believe those both extra details should be immediately displayed there in Version for all builds except the official stable releases!

Twice so since there is "Report an issue" button which just copies that data verbatim.

### PR DESCRIPTION
**Pull request by XVilka**
# SQUASH ME

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [ ] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

Show git branch name and short revision (hash) when available

**Test plan (required)**

Screenshot 2023-02-07 at 11 43 46

**Closing issues**

Closes https://github.com/rizinorg/cutter/issues/3089

### CODE REVIEW

**Comment by wargio** [comment]
i'm ok with this, but let check the others.
@thestr4ng3r @kazarmy @ITAYC0HEN

File: CMakeLists.txt [hunk 1]
```diff
@@ -38,9 +38,27 @@ set(CUTTER_VERSION_MAJOR 2)
...
+# Check whether we got any revision (which isn't
+# always the case, e.g. when someone downloaded a zip file
+if ("${GIT_REV}" STREQUAL "")
+	set(CUTTER_VERSION_FULL "${CUTTER_VERSION}")
+else()
+    execute_process(
+        COMMAND git rev-parse --abbrev-ref HEAD
+        OUTPUT_VARIABLE GIT_BRANCH)
+	string(STRIP "${GIT_REV}" GIT_REV)
+    string(SUBSTRING "${GIT_REV}" 1 7 GIT_REV)
```
Comment (kazarmy) [inline CMakeLists.txt:56]: Why is it `1 7` and not `0 7`?

Comment (XVilka) [reply CMakeLists.txt:56]: @kazarmy because the first character is `'`, if use 0 7 it would have been `'2dc362` instead

Comment (kazarmy) [reply CMakeLists.txt:56]: Oh my version of `git log` doesn't show this `'` so could be an issue but anyway the output is easy to check

**Review by kazarmy — CHANGES_REQUESTED** [review]
The indentation's a bit off. Other than that, lgtm.

File: CMakeLists.txt [hunk 2]
```diff
@@ -38,9 +38,27 @@ set(CUTTER_VERSION_MAJOR 2)
...
+if ("${GIT_REV}" STREQUAL "")
+	set(CUTTER_VERSION_FULL "${CUTTER_VERSION}")
+else()
+    execute_process(
+        COMMAND git rev-parse --abbrev-ref HEAD
+        OUTPUT_VARIABLE GIT_BRANCH)
+	string(STRIP "${GIT_REV}" GIT_REV)
+	string(SUBSTRING "${GIT_REV}" 0 7 GIT_REV)
+	string(STRIP "${GIT_BRANCH}" GIT_BRANCH)
+	set(CUTTER_VERSION_FULL "${CUTTER_VERSION}-${GIT_BRANCH}-${GIT_REV}")
```
Comment (kazarmy) [inline CMakeLists.txt]:
```suggestion
    set(CUTTER_VERSION_FULL "${CUTTER_VERSION}")
else()
    execute_process(
        COMMAND git rev-parse --abbrev-ref HEAD
        OUTPUT_VARIABLE GIT_BRANCH)
    string(STRIP "${GIT_REV}" GIT_REV)
    string(SUBSTRING "${GIT_REV}" 0 7 GIT_REV)
    string(STRIP "${GIT_BRANCH}" GIT_BRANCH)
    set(CUTTER_VERSION_FULL "${CUTTER_VERSION}-${GIT_BRANCH}-${GIT_REV}")
```

**Review by kazarmy — CHANGES_REQUESTED** [review]
The indentation's a bit off. Other than that, lgtm.

**Review by kazarmy — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Comment by kazarmy** [comment]
Just realized that `GIT_BRANCH` and `GIT_REV` don't update on _every_ build ala `gittip` in Rizin, but then we don't encourage users to run Cutter from git.

**Pull request merged**